### PR TITLE
fix: Add missing margin to TextField and Select

### PR DESCRIPTION
**Pull request by dequejosie**
Ref: https://github.com/dequelabs/cauldron/issues/538

### CODE REVIEW

File: packages/styles/forms.css [hunk 1]
```diff
@@ -168,7 +168,11 @@ textarea.Field--has-error:focus:hover,
   display: block;
   position: relative;
   box-sizing: border-box;
-  margin-bottom: var(--space-smallest);
+  margin-bottom: var(--space-small);
```
Comment (dequejosie) [inline packages/styles/forms.css:171]: Having discussed with Aaron, we decided to go with 16px

File: packages/styles/forms.css [hunk 2]
```diff
@@ -168,7 +168,11 @@ textarea.Field--has-error:focus:hover,
...
+}
+
+.Field .Error {
+  min-height: unset;
```
Comment (dequejosie) [inline packages/styles/forms.css:175]: Error has its min-height for 24px, which makes the gap between erroring TextField or Select too large.

File: packages/styles/forms.css [hunk 3]
```diff
@@ -210,7 +218,7 @@ textarea.Field--has-error:focus:hover,
   text-align: left;
   color: var(--field-label-text-color);
   font-size: var(--text-size-small);
-  font-weight: var(--font-weight-bold);
+  font-weight: var(--font-weight-medium);
```
Comment (dequejosie) [inline packages/styles/forms.css:221]: Not directly related to this ticket, but Bobby noticed the font-weight for Field__label was off.

File: packages/styles/forms.css [hunk 4]
```diff
@@ -193,6 +197,10 @@ textarea.Field--has-error:focus:hover,
   margin-bottom: var(--space-smallest);
 }
 
+.Checkbox .Error {
+  min-height: unset;
```
Comment (dequejosie) [inline packages/styles/forms.css:201]: Not directly related to this ticket, but the `Error` margin issue also applies to `Checkbox`.

File: packages/styles/forms.css (hunk 2 above)
Comment (thuey) [reply packages/styles/forms.css:175]: @schne324 @scurker Do you think we could just get rid of the min-height from the `.Error` class? Looking at walnut and the extension, we only ever conditionally render the element that has an error on it (e.g. `{error && <div className="Error">Something went wrong</div>}`

**Review by thuey — APPROVED** [review]

**Pull request merged**